### PR TITLE
adds checking for unique assay id and sample on the client side

### LINKED ISSUE #79: fix logic to match new plate run to one that exists on DB already
when a user uploads new assay results the update_assay will only filter to the current plate run, we need to figure out a way to find a matching one on the database already so that we can do genetic id on

### PR DESCRIPTION
**Pull request by ergz**
a few changes:

1. moved the layout details processing to within the `process_sherlock` function, this was already in one of your pr's but moved it here since that one I think got messed up somehow with git
2. check for duplicate values when inserting raw data, I also ignore when the value is "NTC", we might need to be a bit more general about the "NTC" value since we let them change this around, we should ask if they will actually ever use anything other than "NTC"
3. adds some user verification when trying to insert a plate run that has values that exists in db already.
4. adds a `get_plate_run` function for retrieving existing plate runs

closes #79

### CODE REVIEW

File: R/load-sherlock.R [hunk 1]
```diff
@@ -9,26 +9,27 @@
 #' assay on a plate will have its own control blanks and threshold value.
 #' @returns a table containing thresholds for an event, to be passed to `update_assay_detections()`
 #' @export
-generate_threshold <- function(con, plate_run_identifier, .control_id="NTC") {
+generate_threshold <- function(con, plate_run, .control_id="NTC") {
```
Comment (liz-fw) [inline R/load-sherlock.R:12]: do we need to update the documentation here?

File: R/load-sherlock.R [hunk 2]
```diff
@@ -212,12 +294,12 @@ add_raw_assay_results <- function(con, assay_results) {
 #' database is updated with the genetic identification value. The genetic
 #' identification values are dependent on the assay results:
 #' 1: high value for
-add_genetic_identification <- function(con, sample_identifiers, plate_run_id) {
+add_genetic_identification <- function(con, sample_identifiers) {
```
Comment (liz-fw) [inline R/load-sherlock.R:297]: do we need to update the documentation (no plate_run_id)?

**Review by liz-fw — APPROVED** [review]
looks good to me! i added comments on 2 functions where you change the plate_run_id arg and the docs above don't match

**Comment by ergz** [comment]
yeah, I will also update the docs on website with the new embedded sample layout call

**Pull request merged**